MPS fails developer build on clang 15

I was wondering how to get more coverage of arm64 and realised I had [Termux](https://termux.dev/en/) installed on my phone.

Dataflow analysis in the hot build of mps.c throws a couple of warnings for this particular version of Clang.

```
~/git/mps/code $ uname -a
Linux localhost 4.9.218+ #1 SMP PREEMPT Mon Dec 7 20:15:05 UTC 2020 aarch64 Android
~/git/mps/code $ cc -dumpmachine
aarch64-unknown-linux-android24
~/git/mps/code $ cc --version
clang version 15.0.7
Target: aarch64-unknown-linux-android24
Thread model: posix
InstalledDir: /data/data/com.termux/files/usr/bin
~/git/mps/code $ make -f lia6ll.gmk
make[1]: Entering directory '/storage/emulated/0/git/mps/code'
make[2]: Entering directory '/storage/emulated/0/git/mps/code'
lia6ll: lia6ll/hot/mps.o
In file included from mps.c:88:
./poolawl.c:1060:9: error: variable 'preservedInPlaceCount' set but not used
      [-Werror,-Wunused-but-set-variable]
  Count preservedInPlaceCount = (Count)0;
        ^
In file included from mps.c:89:
./poollo.c:316:9: error: variable 'preservedInPlaceCount' set but not used
      [-Werror,-Wunused-but-set-variable]
  Count preservedInPlaceCount = (Count)0;
        ^
2 errors generated.
make[2]: *** [comm.gmk:615: lia6ll/hot/mps.o] Error 1
make[2]: Leaving directory '/storage/emulated/0/git/mps/code'
make[1]: *** [comm.gmk:395: target] Error 2
make[1]: Leaving directory '/storage/emulated/0/git/mps/code'
make: *** [comm.gmk:366: mps.a] Error 2
```

The deployment build `cc -c mps.c` is fine.

> In file included from mps.c:88:
> ./poolawl.c:1060:9: error: variable 'preservedInPlaceCount' set but not used
>       [-Werror,-Wunused-but-set-variable]
>   Count preservedInPlaceCount = (Count)0;
>         ^

Although `preservedInPlaceCount` is incremented here https://github.com/Ravenbrook/mps/blob/e179634de028b2db090ee6a2358391155102b74f/code/poolawl.c#L1092-L1093
it is only used inside a `STATISTIC` macro here
https://github.com/Ravenbrook/mps/blob/e179634de028b2db090ee6a2358391155102b74f/code/poolawl.c#L1111
and the compiler has noticed.

--- a/code/amcss.c
+++ b/code/amcss.c
@@ -95,13 +95,12 @@ static void report(void)
 
 static mps_addr_t make(size_t rootsCount)
 {
-  static unsigned long calls = 0;
+  static volatile unsigned long calls = 0; /* for use from debugger */
   size_t length = rnd() % (scale * avLEN);
   size_t size = (length+2) * sizeof(mps_word_t);
   mps_addr_t p;
   mps_res_t res;
   ++ calls;
-  (void)calls; /* suppress unused warning: we want to look at this from debugger */
 
   do {
     MPS_RESERVE_BLOCK(res, p, ap, size);

--- a/code/amcss.c
+++ b/code/amcss.c
@@ -101,6 +101,7 @@ static mps_addr_t make(size_t rootsCount)
   mps_addr_t p;
   mps_res_t res;
   ++ calls;
+  (void)calls; /* suppress unused warning: we want to look at this from debugger */
 
   do {
     MPS_RESERVE_BLOCK(res, p, ap, size);

--- a/code/amcss.c
+++ b/code/amcss.c
@@ -95,7 +95,12 @@ static void report(void)
 
 static mps_addr_t make(size_t rootsCount)
 {
-  static volatile unsigned long calls = 0; /* for use from debugger */
+  /* The calls variable is useful when debugging to stop the debugging
+     after a certain number of allocations using a debugger
+     "watchpoint".  Allocations are a good "clock tick" for this test.
+     The test itself doesn't use the variable, so we declare it
+     volatile, which also forces updates into memory. */
+  static volatile unsigned long calls = 0;
   size_t length = rnd() % (scale * avLEN);
   size_t size = (length+2) * sizeof(mps_word_t);
   mps_addr_t p;

--- a/code/poolawl.c
+++ b/code/poolawl.c
@@ -1057,7 +1057,7 @@ static void awlSegReclaim(Seg seg, Trace trace)
   Bool hasBuffer = SegBuffer(&buffer, seg);
   Format format = pool->format;
   Count reclaimedGrains = (Count)0;
-  Count preservedInPlaceCount = (Count)0;
+  STATISTIC_DECL(Count preservedInPlaceCount = (Count)0)
   Size preservedInPlaceSize = (Size)0;
   Index i;
 
@@ -1089,7 +1089,7 @@ static void awlSegReclaim(Seg seg, Trace trace)
       AVER(BTGet(awlseg->scanned, i));
       BTSetRange(awlseg->mark, i, j);
       BTSetRange(awlseg->scanned, i, j);
-      ++preservedInPlaceCount;
+      STATISTIC(++preservedInPlaceCount);
       preservedInPlaceSize += AddrOffset(p, q);
     } else {
       BTResRange(awlseg->mark, i, j);

--- a/code/poollo.c
+++ b/code/poollo.c
@@ -313,7 +313,7 @@ static void loSegReclaim(Seg seg, Trace trace)
   Bool hasBuffer = SegBuffer(&buffer, seg);
   Count reclaimedGrains = (Count)0;
   Format format = NULL; /* supress "may be used uninitialized" warning */
-  Count preservedInPlaceCount = (Count)0;
+  STATISTIC_DECL(Count preservedInPlaceCount = (Count)0)
   Size preservedInPlaceSize = (Size)0;
   Bool b;
 
@@ -355,7 +355,7 @@ static void loSegReclaim(Seg seg, Trace trace)
     q = (*format->skip)(AddrAdd(p, format->headerSize));
     q = AddrSub(q, format->headerSize);
     if(BTGet(loseg->mark, i)) {
-      ++preservedInPlaceCount;
+      STATISTIC(++preservedInPlaceCount);
       preservedInPlaceSize += AddrOffset(p, q);
     } else {
       Index j = PoolIndexOfAddr(base, pool, q);

